src/us/kbase/workspace/database/mongo/MongoWorkspaceDB.java

```diff
@@ -2322,20 +2324,18 @@ private void checkTotalFileSize(
 
 	private void cleanUpTempObjectFiles(
 			final Map<String, ByteArrayFileCache> chksumToData,
-			final Map<ObjectIDResolvedWS, Map<SubsetSelection,
-				WorkspaceObjectData>> ret) {
+			final Map<ObjectIDResolvedWS, Map<SubsetSelection, WorkspaceObjectData.Builder>> ret) {
 		for (final ByteArrayFileCache f: chksumToData.values()) {
 			try {
 				f.destroy();
 			} catch (RuntimeException | Error e) {
 				//continue
 			}
 		}
-		for (final Map<SubsetSelection, WorkspaceObjectData> m:
-			ret.values()) {
-			for (final WorkspaceObjectData wod: m.values()) {
+		for (final Map<SubsetSelection, WorkspaceObjectData.Builder> m: ret.values()) {
+			for (final WorkspaceObjectData.Builder wod: m.values()) {
 				try {
-					wod.destroy();
+					wod.build().destroy();
```

Comment:
Not covered by tests!

Reply:
yeah, this method is going to be completely rewritten shortly so I'm not going to worry about it for now